From df60b547de99c7c217f30fa665ee773328434857 Mon Sep 17 00:00:00 2001
From: Mingcong Bai <jeffbai@aosc.io>
Date: Sun, 6 Sep 2026 03:12:17 +0800
Subject: [PATCH 3/3] AOSCOS: fix(RecordingModeMenuButton): hide the recording
 button outright if not supported

Why show something like "not implemented yet" when there is simply no
implementation available to the user?

Signed-off-by: Mingcong Bai <jeffbai@aosc.io>
---
 src/Gui/RecordingModeMenuButton.qml | 5 +----
 1 file changed, 1 insertion(+), 4 deletions(-)

diff --git a/src/Gui/RecordingModeMenuButton.qml b/src/Gui/RecordingModeMenuButton.qml
index 7692088e..92e1af6d 100644
--- a/src/Gui/RecordingModeMenuButton.qml
+++ b/src/Gui/RecordingModeMenuButton.qml
@@ -7,13 +7,10 @@ import QtQuick.Controls as QQC
 import org.kde.spectacle.private
 
 TtToolButton {
-    enabled: SpectacleCore.videoPlatform.supportedRecordingModes !== VideoPlatform.NoRecordingModes
+    visible: SpectacleCore.videoPlatform.supportedRecordingModes !== VideoPlatform.NoRecordingModes
     icon.name: "camera-video"
     text: i18nc("@action select new recording mode", "New Recording")
     down: pressed || RecordingModeMenu.visible
     Accessible.role: Accessible.ButtonMenu
     onPressed: RecordingModeMenu.popup(this)
-    hoverEnabled: true
-    QQC.ToolTip.visible: !enabled && hovered
-    QQC.ToolTip.text: i18nc("@info:tooltip", "This feature is not supported on the current platform")
 }
-- 
2.55.0

